IPython/utils/text.py

```diff
@@ -447,27 +432,30 @@ def strip_email_quotes(text):
         In [4]: strip_email_quotes('> > text\\n> > more\\n> more...')
         Out[4]: '> text\\n> more\\nmore...'
 
-    So if any line has no quote marks ('>') , then none are stripped from any
+    So if any line has no quote marks ('>'), then none are stripped from any
     of them ::
-    
+
         In [5]: strip_email_quotes('> > text\\n> > more\\nlast different')
         Out[5]: '> > text\\n> > more\\nlast different'
     """
     lines = text.splitlines()
-    matches = set()
-    for line in lines:
-        prefix = re.match(r'^(\s*>[ >]*)', line)
-        if prefix:
-            matches.add(prefix.group(1))
+    strip_len = 0
+
+    for characters in zip(*lines):
+        # Check if all characters in this position are the same
+        if len(set(characters)) > 1:
+            break
+        prefix_char = characters[0][0]
```

Comment:
```suggestion
        prefix_char = characters[0]
```
I guess ? 

If characters is `('a', 'e', 'i', 'o', 'u')`, then just a single `[0]` is needed. The second 0, is just because you index the first char of a string, which is the identity, as we have a tuple of strings of len 1.

unless I misunderstood.

Reply:
Yes, you're right, I made a thinko here. Thanks!